### PR TITLE
Adds a Minimum Code Matter Example

### PR DESCRIPTION
**Pull request by SuGlider**
## Description of Change
Adds the minimum Matter Accessory example code in order to ilustrate how simple it is to work with the new ESP32 Arduino Matter Library.

## Tests scenarios
ESP32-S3 with the provided example

## Related links
Related to #7432

### CODE REVIEW

**Comment by me-no-dev** [comment]
this is not a working example, right? Maybe add the basic callback to it only work so it can change the led builtin (no preferences, etc.)

**Comment by SuGlider** [comment]
> this is not a working example, right? Maybe add the basic callback to it only work so it can change the led builtin (no preferences, etc.)

Done, tested and working.

File: libraries/Matter/examples/MatterMinimum/MatterMinimum.ino [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * This example is the smallest code that will create a Matter Device which can be
+ * commissioned and controlled from a Matter Environment APP.
+ * It doesn't control any light, but the ESP32 will send debug messages indicating the Matter activity.
```
Comment (SuGlider) [inline libraries/Matter/examples/MatterMinimum/MatterMinimum.ino:18]:
```suggestion
 * It controls a GPIO that may be used as LED, additionally the ESP32 will send debug messages indicating the Matter activity.
```

File: libraries/Matter/examples/MatterMinimum/MatterMinimum.ino [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * Therefore, turning DEBUG Level may be useful to follow Matter Accessory and Controller messages.
```
Comment (SuGlider) [inline libraries/Matter/examples/MatterMinimum/MatterMinimum.ino:19]:
```suggestion
 * Turning DEBUG Level may be useful to follow Matter Accessory and Controller messages.
```

File: libraries/Matter/examples/MatterMinimum/MatterMinimum.ino [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * This example is the smallest code that will create a Matter Device which can be
+ * commissioned and controlled from a Matter Environment APP.
+ * It controls a GPIO that may be used as LED, additionally the ESP32 will send debug messages indicating the Matter activity.
```
Comment (SuGlider) [inline libraries/Matter/examples/MatterMinimum/MatterMinimum.ino:18]:
```suggestion
 * It controls a GPIO, that could be attached to a LED. Additionally the ESP32 will send debug messages indicating the Matter activity.
```

File: libraries/Matter/examples/MatterMinimum/MatterMinimum.ino [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * This example is the smallest code that will create a Matter Device which can be
+ * commissioned and controlled from a Matter Environment APP.
+ * It controls a GPIO, that could be attached to a LED. Additionally the ESP32 will send debug messages indicating the Matter activity.
```
Comment (SuGlider) [inline libraries/Matter/examples/MatterMinimum/MatterMinimum.ino:18]:
```suggestion
 * It controls a GPIO that could be attached to a LED for visualization.
 * Additionally the ESP32 will send debug messages indicating the Matter activity.
```

**Review by lucasssvaz — APPROVED** [review]

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**